Jolt/Math/Vec3.inl

```diff
@@ -527,67 +533,87 @@ Vec3 Vec3::Cross(Vec3Arg inV2) const
 
 Vec3 Vec3::DotV(Vec3Arg inV2) const
 {
-#if defined(JPH_USE_SSE)
+#if defined(JPH_USE_SSE4_1)
 	return _mm_dp_ps(mValue, inV2.mValue, 0x7f);
 #elif defined(JPH_USE_NEON)
     float32x4_t mul = vmulq_f32(mValue, inV2.mValue);
 	mul = vsetq_lane_f32(0, mul, 3);
     return vdupq_n_f32(vaddvq_f32(mul));
 #else
-	#error Unsupported CPU architecture
+	float dot = 0.0f;
+	for (int i = 0; i < 3; i++) {
+		dot += mF32[i] * inV2.mF32[i];
+	}
+	return Vec3(dot, dot, dot);
 #endif
 }
 
 Vec4 Vec3::DotV4(Vec3Arg inV2) const
 {
-#if defined(JPH_USE_SSE)
+#if defined(JPH_USE_SSE4_1)
 	return _mm_dp_ps(mValue, inV2.mValue, 0x7f);
 #elif defined(JPH_USE_NEON)
     float32x4_t mul = vmulq_f32(mValue, inV2.mValue);
 	mul = vsetq_lane_f32(0, mul, 3);
     return vdupq_n_f32(vaddvq_f32(mul));
 #else
-	#error Unsupported CPU architecture
+	float dot = 0.0f;
+	for (int i = 0; i < 3; i++) {
+		dot += mF32[i] * inV2.mF32[i];
+	}
+	return Vec4(dot, dot, dot, dot);
 #endif
 }
 
 float Vec3::Dot(Vec3Arg inV2) const
 {
-#if defined(JPH_USE_SSE)
+#if defined(JPH_USE_SSE4_1)
 	return _mm_cvtss_f32(_mm_dp_ps(mValue, inV2.mValue, 0x7f));
 #elif defined(JPH_USE_NEON)
     float32x4_t mul = vmulq_f32(mValue, inV2.mValue);
 	mul = vsetq_lane_f32(0, mul, 3);
     return vaddvq_f32(mul);
 #else
-	#error Unsupported CPU architecture
+	float dot = 0.0f;
+	for (int i = 0; i < 3; i++) {
+		dot += mF32[i] * inV2.mF32[i];
+	}
+	return dot;
 #endif
 }
 
 float Vec3::LengthSq() const
 {
-#if defined(JPH_USE_SSE)
+#if defined(JPH_USE_SSE4_1)
 	return _mm_cvtss_f32(_mm_dp_ps(mValue, mValue, 0x7f));
 #elif defined(JPH_USE_NEON)
     float32x4_t mul = vmulq_f32(mValue, mValue);
 	mul = vsetq_lane_f32(0, mul, 3);
     return vaddvq_f32(mul);
 #else
-	#error Unsupported CPU architecture
+	float len_sq = 0.0f;
+	for (int i = 0; i < 3; i++) {
+		len_sq += mF32[i] * mF32[i];
+	}
+	return len_sq;
 #endif
 }
 
 float Vec3::Length() const
 {
-#if defined(JPH_USE_SSE)
+#if defined(JPH_USE_SSE4_1)
 	return _mm_cvtss_f32(_mm_sqrt_ss(_mm_dp_ps(mValue, mValue, 0x7f)));
 #elif defined(JPH_USE_NEON)
     float32x4_t mul = vmulq_f32(mValue, mValue);
 	mul = vsetq_lane_f32(0, mul, 3);
     float32x2_t sum = vdup_n_f32(vaddvq_f32(mul));
     return vget_lane_f32(vsqrt_f32(sum), 0);
 #else
-	#error Unsupported CPU architecture
+	float len_sq = 0.0f;
+	for (int i = 0; i < 3; i++) {
+		len_sq += mF32[i] * mF32[i];
+	}
+	return sqrtf(len_sq);
```

Comment:
Jolt uses the C++ sqrt function instead of the C sqrtf function. Can you change that? (happens multiple times)